app/stores/base/Store.ts

```diff
@@ -144,6 +149,42 @@ export default abstract class Store<T extends Model> {
     LifecycleManager.executeHooks(model.constructor, "afterRemove", model);
   }
 
+  @action
+  addToArchive(item: PartialWithArchivedAt<T>): void {
```

Comment:
I think there should be a check if the model has `archivedAt` property at all, and if not throw an error as a sanity check.

Reply:
That's fine too